lib/ex_doc/autolink.ex

```diff
@@ -501,7 +482,9 @@ defmodule ExDoc.Autolink do
 
         nil
 
-      {:regular_link, _module_visibility, :undefined} when not same_module? ->
+      {:regular_link, _module_visibility, :undefined}
+      when not same_module? and
+             (config.language != ExDoc.Language.Erlang or kind == :function) ->
```

Comment:
Same here, feel free to let it warn on Elixir, assuming it is one of `m:`, `t:` or `c:`.